envsec/internal/auth/aws.go

```diff
@@ -0,0 +1,155 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/MicahParks/keyfunc/v2"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cognitoidentity"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/golang-jwt/jwt/v5"
+	"go.jetpack.io/envsec/debug"
+)
+
+// for valiation of JWT (can be hardcoded to avoid fetching)
+const jwksURL = `https://jetpack-io.us.auth0.com/.well-known/jwks.json`
+const awsRegion = "us-east-1"
+
+// have to be var instead of const so that their &address can be passed
+var accountId = "984256416385"
+var identityPoolId = "us-east-1:da3c3c71-61c7-4f7c-8e3d-3770e9b61379"
```

Comment:
One thing I find helpful is to declare a config struct and then put hard coded stuff there. That way once we stop hardcoding it's easy to replace with env vars or build vars etc.